native/src/main/java/io/ballerina/stdlib/ftp/server/FtpListener.java

```diff
@@ -129,6 +102,44 @@ public void notifyFailure(BError error) {
         return true;
     }
 
+    private Object[] getMethodArguments(Parameter[] params, Map<String, Object> watchEventParamValues) {
+        if (params.length == 1) {
+            return new Object[] {getWatchEvent(params[0], watchEventParamValues), true};
+        } else if (params.length == 2) {
+            if ((params[0].type.isReadOnly() || params[0].type.getTag() == RECORD_TYPE_TAG) &&
+                    params[1].type.getTag() == OBJECT_TYPE_TAG) {
+                return new Object[] {getWatchEvent(params[0], watchEventParamValues), true, caller, true};
+            } else if ((params[1].type.isReadOnly() || params[1].type.getTag() == RECORD_TYPE_TAG) &&
+                    params[0].type.getTag() == OBJECT_TYPE_TAG) {
+                return new Object[] {caller, true, getWatchEvent(params[1], watchEventParamValues), true};
+            } else {
+                log.error("Invalid parameter types in onFileChange method");
+            }
+        } else {
+            log.error("Invalid parameter count in onFileChange method");
```

Comment:
Just to get an idea. Haven't we done these two validations at the compiler plugin? 
(No need of removing logs)

Reply:
yeah. both the validations are there